src/gflags.h.in

```diff
@@ -429,11 +429,57 @@ extern GFLAGS_DLL_DECL void ShutDownCommandLineFlags();
 // people can't DECLARE_int32 something that they DEFINE_bool'd
 // elsewhere.
 
+enum FlagValueType {
+  FVT_BOOL = 0,
+  FVT_INT32 = 1,
+  FVT_UINT32 = 2,
+  FVT_INT64 = 3,
+  FVT_UINT64 = 4,
+  FVT_DOUBLE = 5,
+  FVT_STRING = 6,
+  FVT_MAX_INDEX = 6
+};
+
+template <typename T, const FlagValueType type>
+class FlagTraitsBase {
+ public:
+  typedef T ValueType;
+
+  static FlagValueType Type() { return type; }
+};
+
+template <typename T>
+class FlagTraits;
+
+template <>
+class FlagTraits<bool> : public FlagTraitsBase<bool, FVT_BOOL> {};
+template <>
+class FlagTraits<int32> : public FlagTraitsBase<int32, FVT_INT32> {};
+template <>
+class FlagTraits<int64> : public FlagTraitsBase<int64, FVT_INT64> {};
+template <>
+class FlagTraits<uint32> : public FlagTraitsBase<uint32, FVT_UINT32> {};
+template <>
+class FlagTraits<uint64> : public FlagTraitsBase<uint64, FVT_UINT64> {};
+template <>
+class FlagTraits<double> : public FlagTraitsBase<double, FVT_DOUBLE> {};
+template <>
+class FlagTraits<std::string> : public FlagTraitsBase<std::string, FVT_STRING> {
+};
+
 class GFLAGS_DLL_DECL FlagRegisterer {
  public:
-  FlagRegisterer(const char* name, const char* type,
+  FlagRegisterer(const char* name, FlagValueType type,
                  const char* help, const char* filename,
                  void* current_storage, void* defvalue_storage);
+
+  template <typename T>
+  FlagRegisterer(const char* name, const char* help, const char* filename,
```

Comment:
This is a type-safe helper ctor, and it would be better with C++11 feature `delegating constructors` =(
I used a temporary object, because FlagRegisterer don't have any fields.